pkg/daemon/controller/nodeinfo.go

```diff
@@ -345,22 +376,20 @@ func (r *nodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return fmt.Errorf("failed to create node controller: %v", err)
 	}
 
-	if err := nodeController.Watch(&source.Kind{Type: &corev1.Node{}},
-		&fixedKeyHandler{key: ActionReconcileNode},
-		&predicate.ResourceVersionChangedPredicate{},
-		&predicate.AnnotationChangedPredicate{},
-		&predicate.Funcs{
-			UpdateFunc: checkNodeUpdate,
-		},
+	if err := nodeController.Watch(&source.Kind{Type: &networkingv1.NodeInfo{}},
+		&fixedKeyHandler{key: ActionReconcileNodeInfo},
+		&predicate.GenerationChangedPredicate{},
 	); err != nil {
 		return fmt.Errorf("failed to watch corev1.Node for node controller: %v", err)
 	}
 
 	if err := nodeController.Watch(&source.Kind{Type: &networkingv1.Network{}},
-		&fixedKeyHandler{key: ActionReconcileNode},
+		&fixedKeyHandler{key: ActionReconcileNodeInfo},
 		predicate.Funcs{
 			UpdateFunc: func(updateEvent event.UpdateEvent) bool {
-				return false
+				oldNetwork := updateEvent.ObjectOld.(*networkingv1.Network)
+				newNetwork := updateEvent.ObjectNew.(*networkingv1.Network)
+				return !utils2.StringSliceEqual(oldNetwork.Status.NodeList, newNetwork.Status.NodeList)
```

Comment:
if `network.Status.NodeList` changed, we believe `NodeInfo` will trigger a valid event, so here should always return false.

Reply:
This is for the case that `NodeInfo` is deleted manually first and then `Node` be deleted. In this case, the event triggered by `NodeInfo` will not cause the delete operation of vxlan fdb entries on other nodes. If the next `Node` deletion step will not cause an event either, vxlan fdb entries on other nodes will not be cleaned correctly.